### PR TITLE
[Hot Fix] Give priority to plugins to set distributed mode, and then accelerator

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Fixes the broken API case where:

```python
Trainer(gpus=1, plugins='ddp_sharded', accelerator='ddp')
```

Because of the logic within the accelerator connector, the plugin is never set correctly, and the accelerator takes priority. We reverse the priority, and I'll make an issue for a RFC to fix this API.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed incorrect yield logic for the amp autocast context manager ([#6080](https://github.com/PyTorchLightning/pytorch-lightning/pull/6080))
 
 
+- Fixed priority of plugin/accelerator when setting distributed mode ([#6089](https://github.com/PyTorchLightning/pytorch-lightning/pull/6089))
+
```
Comment (carmocca) [inline CHANGELOG.md:28]:
```suggestion


```

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
